listenbrainz/webserver/views/api_tools.py

```diff
@@ -104,6 +106,12 @@ def validate_listen(listen, listen_type):
             log_raise_400("JSON document may only contain listened_at and "
                            "track_metadata top level keys", listen)
 
+        # if timestamp is too high, raise BadRequest
+        # in order to make up for possible clock skew, we allow
+        # timestamps to be one hour ahead of server time
+        if listen['listened_at'] > int(time.time()) + API_LISTENED_AT_ALLOWED_SKEW:
```

Comment:
Sorry, I should've seen this earlier, but this allows for only a clock skew for the users clock to be FAST, not SLOW. I think we should test for 30 minutes fast and or 30 minutes slow.

Reply:
If I do  `current_time - API_LISTENED_AT_ALLOWED_SKEW <= ts <= currentt_time + API_LISTENED_AT_ALLOWED_SKEW`, it wouldn't work for old listens leading to failing imports.

Reply:
Sorry, can you please elaborate? I don't understand what you're saying.